tests/utils/test_url.py

```diff
@@ -0,0 +1,145 @@
+from unittest.mock import MagicMock, patch
+
+import requests
+from page_analyzer.utils.url import (
+    check_url,
+    normalize,
+    parse_page_data,
+    validate,
+)
+
+
+def test_validate():
+    # given
+    url = "https://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {}
+
+
+def test_validate_invalid():
+    # given
+    url = "htttps://www.weforum.org/my-forum/"
+
+    # when
+    errors = validate(url)
+
+    # then
+    assert errors == {"name": "Некорректный URL"}
+
+
+def test_normalize():
+    # given
+    url = "http://www.example.com:1030/software/index.html"
+
+    # when
+    normalized_url = normalize(url)
+
+    # then
+    assert normalized_url == "http://www.example.com"
+
+
+def test_check_url():
+    # given
+    url = "https://example.com"
+    parsing_data = {
+        "response_code": "200",
+        "h1": "some header",
+        "title": "some title",
+        "description": "some description",
+    }
+    expected_html_content = (
+        "<title>some title</title>"
+        "<meta name='description' content='some description'></meta>"
+        "<h1>some header</h1>"
+    )
+    request_response_mock = MagicMock()
+    request_response_mock.status_code = 200
+
+    # when
+    with patch(
+        "page_analyzer.utils.url.requests.get", return_value=request_response_mock
+    ) as mock_requests:
+        request_response_mock.content = expected_html_content
+
+        # then
+        expected_parsing_data = parsing_data
+        assert check_url(url) == expected_parsing_data
+        mock_requests.assert_called_once_with(url)
+
+
+def test_check_url_exception():
+    # given
+    url = "http://www.fake.com"
+
+    # when
+    with patch(
+        "page_analyzer.utils.url.requests.get",
+        side_effect=requests.exceptions.RequestException,
+    ) as mock_requests:
+        # then
+        assert check_url(url) is None
+        mock_requests.assert_called_once_with(url)
+
+
+def test_check_url_unavailable():
+    # given
+    url = "http://www.unavailable.com"
+
+    request_response_mock = MagicMock()
+    request_response_mock.status_code = 404
+
+    # when
+    with patch(
+        "page_analyzer.utils.url.requests.get", return_value=request_response_mock
+    ) as mock_requests:
+        mock_requests.get.return_value = request_response_mock
```

Comment:
is it necessary to assign a return value to the `get` property of the patch?

Reply:
forgot to remove